app.py

```diff
@@ -87,8 +63,73 @@ def create_list():
                 temp_p = temp_p.next
```

Comment:
what the purpose of 
 `  while temp_p and start_of_loop == list.head:
                if random.randint(0, 1):
                    start_of_loop = temp_p
                temp_p = temp_p.next
`